src/main/java/frc/robot/commands/auto/AutoShoot.java

```diff
@@ -0,0 +1,92 @@
+/* ==================================================
+ * Authors: Lucas Jacobs
+ *
+ * --------------------------------------------------
+ * Description:
+ * Makes the robot shoot a ball at the target
+ * ================================================== */
+
+package frc.robot.commands.auto;
+
+import frc.robot.Robot;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class AutoShoot extends CommandBase
+{
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+
+  // VARIABLES //
+
+  private Robot robot;
+  private Timer timer;
+  private double speed;
+  private double timeToShot;
+  boolean finished;
+  private final static double conveyorSpinTime = 1.5;
+
+  // CONSTANTS //
+  final static double TURN_SPEED = 0.1; //CHANGE THIS ALONG WITH VISION PACKAGE
+
+  // CONSTRUCTOR //
+
+  public AutoShoot(Robot robot, double t, double speed)
+  {
+    this.robot = robot;
+    this.speed = speed;
+    timer = new Timer();
+    timeToShot = t;
+    finished = false;
+
+    // subsystems that this command requires
+    addRequirements(robot.vision, robot.drivetrain, robot.shooter, robot.conveyor);
+  }
+
+  // METHODS //
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize()
+  {
+    timer.start();
+    SmartDashboard.putString("Auto Mode:", "Shoot");
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute()
+  {
+
+    //Spin shoot
+    double autoSpeed = robot.vision.autoShooterSpeed();
+    //robot.shooter.spin(autoSpeed);
+    robot.shooter.spin(speed);
+
+    //Aim or Shoot
+    if (timer.get() < timeToShot) {
+      // turn = robot.vision.aimAtTarget(); //Get the turn speed from the camera
+
+      //robot.drivetrain.moveMotors(0, turn * TURN_SPEED);
+    } else if (timer.get() < timeToShot + conveyorSpinTime) {
+
+      robot.conveyor.feed();
+    } else {
+      finished = true;
+    }
```

Comment:
I think you could also write this with `while` loops, not sure if it would cause scheduling issues though.
```
while (timer.get() < timeToShot)
{
    turn drivetrain using vision
}

timer.reset()
while (timer.get() < conveyorSpinTime)
{
    feed conveyor
}
```

Then just exit with `true` at the end.  Although not sure if this would cause issues if you run it in parallel with any other commands.  But I think this is more of a sequential command anyway.

The other thing you could do is break this command up into two simpler ones and run them sequentially.  So something like `AutoAlign()` to turn the drivetrain, then `AutoShooterSpin()` and `ConveyorSpin()` would run in parallel (with a delay on ConveyorSpin() to give the shooter time to charge up).

Reply:
I think the code right now works equally well as your first suggestion, so there's not much point changing it. Also, every time we spin the shooter in auto mode, we will want the conveyor to spin and the robot to align itself, so I don't see the point in splitting this command up.

Reply:
That's a good point I agree.  If it ain't broke don't fix it!